lib/AST/ASTMangler.cpp

```diff
@@ -3838,7 +3838,7 @@ CanType ASTMangler::getDeclTypeForMangling(
 
   auto &C = decl->getASTContext();
   if (decl->isInvalid()) {
```

Comment:
Move this up here: `Type ty = decl->getInterfaceType()->getReferenceStorageReferent();` and replace the `isInvalid()` check with `if (ty->hasError())`